medusa/reconstruct/expand.py

```diff
@@ -1,261 +1,72 @@
-# functions for expanding existing models to generate an ensemble
-
-REACTION_ATTRIBUTES = ['lower_bound', 'upper_bound']
-MISSING_ATTRIBUTE_DEFAULT = {'lower_bound':0,'upper_bound':0}
+def probability_dictionary(file_directory):
+    """
+    This function takes the directory forprobability files 
+    and generate a reaction with their probabilities
+    """
+    reaction_probability = {}
+    probability_data = pd.read_csv(file_directory)
+    header_row = 0
+    probability_data = probability_data.transpose()
+    probability_data.columns = probability_data.iloc[header_row]
+    probability_data  = probability_data.drop('Unnamed: 0')
+    for w in probability_data.index:
+        f = float(probability_data.loc[w, "probability"])
+        reaction_probability[w] = f 
+    return reaction_probability
 
 def gapfill_to_ensemble(model, iterations=1, universal=None, lower_bound=0.05,
-                 penalties=None, exchange_reactions=False,
+                 penalties= None, exchange_reactions=False,
                  demand_reactions=False, integer_threshold=1e-6):
-    """
-    Performs gapfilling on model, pulling reactions from universal.
-    Any existing constraints on base_model are maintained during gapfilling, so
-    these should be set before calling gapfill_to_ensemble (e.g. secretion of
-    metabolites, choice of objective function etc.).
-
-    Currently, only iterative solutions are supported with accumulating
-    penalties (i.e. after each iteration, the penalty for each reaction
-    doubles).
-
-    Parameters
-    ----------
-    model : cobra.Model
-        The model to perform gap filling on.
-    universal : cobra.Model
-        A universal model with reactions that can be used to complete the
-        model.
-    lower_bound : float, 0.05
-        The minimally accepted flux for the objective in the filled model.
-    penalties : dict, None
-        A dictionary with keys being 'universal' (all reactions included in
-        the universal model), 'exchange' and 'demand' (all additionally
-        added exchange and demand reactions) for the three reaction types.
-        Can also have reaction identifiers for reaction specific costs.
-        Defaults are 1, 100 and 1 respectively.
-    integer_threshold : float, 1e-6
-        The threshold at which a value is considered non-zero (aka
-        integrality threshold). If gapfilled models fail to validate,
-        you may want to lower this value. However, picking a threshold that is
-        too low may also result in reactions being added that are not essential
-        to meet the imposed constraints.
-    exchange_reactions : bool, False
-        Consider adding exchange (uptake) reactions for all metabolites
-        in the model.
-    demand_reactions : bool, False
-        Consider adding demand reactions for all metabolites.
-
-    Returns
-    -------
-    ensemble : medusa.core.Ensemble
-        The ensemble object created from the gapfill solutions.
-    """
```

Comment:
Make sure you don't delete these docstrings for any functions--they are automatically parsed to build the documentation at https://medusa.readthedocs.io/en/latest/